packages/gensx/tests/checkpoint.test.tsx

```diff
@@ -357,4 +362,107 @@ suite("checkpoint", () => {
       ],
     });
   });
+
+  test("handles streaming components", async () => {
+    // Define a streaming component that yields tokens with delays
+    const StreamingComponent = gsx.StreamComponent<{ tokens: string[] }>(
+      "StreamingComponent",
+      ({ tokens }) => {
+        const stream = async function* () {
+          for (const token of tokens) {
+            await setTimeout(0); // Small delay between tokens
+            yield token;
+          }
+        };
+        return stream();
+      },
+    );
+
+    // Test non-streaming mode first
+    const { result: nonStreamingResult, checkpoints: nonStreamingCheckpoints } =
+      await executeWithCheckpoints<string>(
+        <StreamingComponent tokens={["Hello", " ", "World"]} stream={false} />,
+      );
+
+    // Verify non-streaming execution
+    expect(nonStreamingResult).toBe("Hello World");
+    const nonStreamingFinal =
+      nonStreamingCheckpoints[nonStreamingCheckpoints.length - 1];
+    expect(nonStreamingFinal).toMatchObject({
+      componentName: "StreamingComponent",
+      props: { tokens: ["Hello", " ", "World"] },
+      output: "Hello World",
+    });
+
+    // Test streaming mode
+    const {
+      result: streamingResult,
+      checkpoints: streamingCheckpoints,
+      checkpointManager,
+    } = await executeWithCheckpoints<AsyncGenerator<string>>(
+      <StreamingComponent tokens={["Hello", " ", "World"]} stream={true} />,
+    );
+
```

Comment:
We should verify the intermediate checkpoint state.

Reply:
I tried doing this, but there is no easy way to do so since intermediate checkpoints are intentionally skipped in the queue (and the skipping is somewhat non-deterministic depending on the timing of the event loop).